src/main/java/seedu/duke/parsers/TagParser.java

```diff
@@ -0,0 +1,47 @@
+package seedu.duke.parsers;
+
+import seedu.duke.commands.Command;
+import seedu.duke.commands.TagCommand;
+import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.exceptions.ParseException;
+import seedu.duke.util.StringConstants;
+
+import java.util.HashMap;
+
+public class TagParser extends Parser {
+    public static final String TAG_OPERATION = StringConstants.TAG_OPERATION;
+    public static final String TASK_NUMBER = StringConstants.TASK_NUMBER;
+    public static final String TASK_MODULE = StringConstants.TASK_MODULE;
+    public static final String TAG_DESCRIPTION = StringConstants.TAG_DESCRIPTION;
+
+    //Unescaped Regex for testing:
+    //((?<taskOperation>\w+))(\s+(?<taskNumber>\d+))((\s+-m\s+(?<taskModule>\w+))?)(\s+(?<tagDescription>.+))
+    private static final String TAG_FORMAT = "((?<tagOperation>\\w+))(\\s+(?<taskNumber>\\d+))"
```

Comment:
Perhaps you can restrict the tagOperation group to add or del to simplify where the exception is thrown